src/ducks/collections.js

```diff
@@ -145,34 +126,40 @@ export const putDocument = (collection, directory, filename) => (
 };
 
 export const deleteDocument = (collection, directory, filename) => dispatch => {
-  return fetch(documentAPIUrl(collection, directory, filename), {
-    method: 'DELETE',
-    credentials: 'same-origin',
-  })
-    .then(data => {
-      dispatch({ type: DELETE_DOCUMENT_SUCCESS });
-      dispatch(fetchCollection(collection, directory));
-    })
-    .catch(error =>
-      dispatch({
-        type: DELETE_DOCUMENT_FAILURE,
-        error,
-      })
-    );
+  return del(
+    documentAPIUrl(collection, directory, filename),
+    {
+      type: DELETE_DOCUMENT_SUCCESS,
+      name: 'doc',
+      update: fetchCollection(collection, directory),
+    },
+    { type: DELETE_DOCUMENT_FAILURE, name: 'error' },
+    dispatch
+  );
 };
 
-const generateFilenameFromTitle = (metadata, collection) => {
+const getFilenameFromTitle = (title, collection, date) => {
+  const slugifiedTitle = slugify(title);
   if (collection === 'posts') {
     // if date is provided, use it, otherwise generate it with today's date
-    let date;
-    if (metadata.date) {
-      date = metadata.date.split(' ')[0];
-    } else {
-      date = moment().format('YYYY-MM-DD');
-    }
-    return `${date}-${slugify(metadata.title)}.md`;
+    const docDate = date ? date.split(' ')[0] : moment().format('YYYY-MM-DD');
+    return `${docDate}-${slugifiedTitle}.md`;
   }
-  return `${slugify(metadata.title)}.md`;
+  return `${slugifiedTitle}.md`;
+};
+
+const validateMetadata = (metadata, collection, directory) => {
```

Comment:
This is not just validating metadata but also mutating `path`. I don't think we should mix things here. In order to understand what this function does to path, we need to look into it. It seems not write to me.

Reply:
So, what would be the best way to proceed yet avoid repeating the following logic in every duck (that handles `metadata`)?
```js
const validateMetadata = (metadata, directory) => {
  let { path, title } = metadata;
  let errors = [];

  if ((!path || `${path}/` === directory) && title) {
    path = getFilenameFromTitle(title);
  } else {
    errors = validatePage(metadata);
  }
  return { path, errors };
};
```